roles/edpm_bootstrap/tasks/fips.yml

```diff
@@ -0,0 +1,44 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Change FIPS status
+  when: edpm_bootstrap_fips_mode != 'check'
```

Comment:
This block should only be run when the detected fips state is different from the requested one. Otherwise there will be an unnecessary reboot. This should be a no-op with a fips enabled image and edpm_bootstrap_fips_mode=enabled

Reply:
+1  Just a question about the  reboot.  When does this reboot occur and what happens before then?  The reason  I ask is because fips mode really isn't enabled until the reboot occurs.  This means that any crypto operations (generating keys for example) that occur won't occur in fips mode until the reboot occurs.

Reply:
+1  Just a question about the  reboot.  When does this reboot occur and what happens before then?  The reason  I ask is because fips mode really isn't enabled until the reboot occurs.  This means that any crypto operations (generating keys for example) that occur won't occur in fips mode until the reboot occurs.

Reply:
Oops, I missed this comment thread. I just removed the lgtm label until this one is addressed.

Reply:
@steveb ack. I will improve the tasks to be executed only if the node is on a different state than the one requested.

@vakwetu the reboot will be executed when the `reboot-os` service is defined while deploying, hence the `reboot` playbook is called. This integration was requested by @slagle (and I 100% support that) and already used by other roles (edpm_kernel)

Reply:
The reason we prefer to set FIPS as early as possible - even to use FIPS enabled images - is to ensure that the right crypto policies are in place before an crypto operations (generation of keys etc) are done.  That way, we are assured that everything is done in a FIPS compliant way.

If we don't reboot right away, then we don't know what other playbooks will be called before the reboot-os service.  Maybe it will all work out and things will still work when fips is properly enabled -- then again, maybe it won't.

Reply:
@vakwetu i get your point, so i enforced the reboot of the node right after the call of setup-fips-mode (both for enabling/desabling mode).

I also add a molecule test on a proper vm, let's see how it goes. :)